pallets/settlement/src/lib.rs

```diff
@@ -1098,10 +1118,8 @@ impl<T: Trait> Module<T> {
         )?;
 
         with_transaction(|| {
-            let legs = <InstructionLegs<T>>::iter_prefix(instruction_id);
-            for (leg_id, leg_details) in
-                legs.filter(|(_leg_id, leg_details)| portfolios.contains(&leg_details.from))
-            {
+            let legs = Self::get_valid_legs(instruction_id, legs_count, &portfolios)?;
```

Comment:
This does no storage changes, let's move this before and out of the transaction

Reply:
I don't think so it will matter even it is better to have this within the transaction so it can free up early from the scope.

Reply:
Moving this before `with_transaction` will mean "freeing up" even earlier, not paying any cost for `with_transaction` initialization at all (there is some cost there). As a general guideline, `with_transaction`s should be as short as possible, only combining those functions which both have "fail or write if ok" semantics.